api/src/main/java/org/hawkular/inventory/api/model/MetricType.java

```diff
@@ -157,7 +155,7 @@ private Update() {
 
         public Update(Map<String, Object> properties, MetricUnit unit) {
             super(properties);
-            this.unit = unit;
+            this.unit = unit == null ? MetricUnit.NONE : unit;
```

Comment:
The above is still true. While you added the code to check for unit's nullity in the backend, in here you're essentially disallowing it ever be null. I think what you did for properties is correct, but unit is still not behaving correctly, IMHO.